src/main/java/sora/task/TaskList.java

```diff
@@ -0,0 +1,286 @@
+            throw new TaskNotFoundException();
+        }
+
+        // Mark the task as done
+        Task task = tasks.get(taskNumber);
+        task.markAsDone();
+
+        // Return confirmation message
+        return Message.DONE + "\n  " + task;
+    }
+
+    /**
+     * Adds a new event to the list of tasks.
+     *
+     * @param command command entered by user (event [description] /at [dd/MM/yy] /from [HHmm] /to [HHmm])
+     * @return confirmation message for successfully added a new event
+     * @throws IllegalFormatException if user inputs an invalid command
+     */
+    public String addEvent(String command) throws IllegalFormatException {
+        String correctFormat = "event [description] /at [dd/MM/yy] /from [HHmm] /to [HHmm]";
+
+        // Throw exception if command does not follow format
+        validateCommand(command, "^event .* /at .* /from .* /to .*", correctFormat);
+
+        // Separate info and trim each of them
+        String[] info = command.substring(6).split("/at|/from|/to");
+        info = Arrays.stream(info)
+            .map(String::trim)
+            .toArray(String[]::new);
+
+        LocalDate date;
+        LocalTime startTime, endTime;
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("Hmm");
+
+        // Throw exception if command does not follow format
+        try {
+            date = LocalDate.parse(info[1], DateTimeFormatter.ofPattern("d/M/yy"));
+            startTime = LocalTime.parse(info[2], timeFormatter);
+            endTime = LocalTime.parse(info[3], timeFormatter);
+        } catch (DateTimeParseException e) {
+            throw new IllegalFormatException(correctFormat);
+        }
+
+        // Throw exception if start time is later than end time
+        if (startTime.isAfter(endTime)) {
+            throw new IllegalFormatException(correctFormat);
+        }
+
+        // Add new event into list of tasks
+        Task newTask = new Event(info[0], date, startTime, endTime);
+        tasks.add(newTask);
+
+        // Return confirmation message
+        return formatAddMessage(newTask, tasks.size());
+    }
+
+    /**
+     * Adds a new deadline to the list of tasks.
+     *
+     * @param command command entered by user (deadline [description] /by [dd/MM/yy] [HHmm])
+     * @return confirmation message for successfully added a new deadline
+     * @throws IllegalFormatException if user inputs an invalid command
+     */
+    public String addDeadline(String command) throws IllegalFormatException {
+        String correctFormat = "deadline [description] /by [dd/MM/yy] [HHmm]";
+
+        // Throw exception if command does not follow format
+        validateCommand(command, "^deadline .* /by .*", correctFormat);
+
+        // Separate info and trim each of them
+        String[] info = command.substring(9).split("/by");
+        info = Arrays.stream(info)
+            .map(String::trim)
+            .toArray(String[]::new);
```

Comment:
Same as above, may need to add 4 space indentation(?)
Happens quite a few times below as well so I'll just comment here 👍